config/routes.rb

```diff
@@ -1,8 +1,14 @@
 Rails.application.routes.draw do
+  get 'dashboard', to: 'dashboard#index'
+  get 'search', to: 'search#index'
+  get 'notifications', to: 'notifications#index'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # https://github.com/heartcombo/devise/blob/main/README.md
-  devise_for :users
+  devise_for :users, controllers: { registrations: 'users/registrations' }
```

Comment:
Could you maybe explain why this change to devise was necessary?

Reply:
We added our own Users controller extending devises registration controller because we wanted to add a profile page to that controller. We did not want to create a new controller for this, as this functionality is tightly coupled with the login / registration process. In order for devise to use our controller instead of the default one for registrations, we needed to change this config.

Reply:
ok